modules/luci-mod-dashboard/htdocs/luci-static/resources/view/dashboard/include/10_router.js

```diff
@@ -0,0 +1,331 @@
+		container_box.appendChild(E('hr'));
+
+		if ('internet' == type) {
+			var container_internet_v4 = E('div');
+			var container_internet_v6 = E('div');
+
+			for(var idx in data) {
+
+				for(var ver in data[idx]) {
+					var classname = ver,
+						suppelements = '',
+						visible = data[idx][ver].visible;
+
+					if('connected' === ver) {
+						classname = data[idx][ver].value ? 'label label-success' : 'label label-danger';
+						data[idx][ver].value = data[idx][ver].value ? _('yes') : _('no');
+						//suppelements = E('i', {}, data[idx].uptime.value);
+					}
+
+					if ('v4' === idx) {
+
+						if ('title' === ver) {
+							container_internet_v4.appendChild(
+								E('p', { 'class': 'mt-2'}, [
+									E('span', {'class': ''}, [ data[idx].title ]),
+								])
+							);
+							continue;
+						}
+
+						if ('addrsv4' === ver) {
+							var addrs = data[idx][ver].value;
+							if(Array.isArray(addrs) && addrs.length) {
+								for(var ip in addrs) {
+									data[idx][ver].value = addrs[ip].split('/')[0];
+								}
+							}
+						}
+
+						if (visible) {
+							container_internet_v4.appendChild(
+								E('p', { 'class': 'mt-2'}, [
+									E('span', {'class': ''}, [ data[idx][ver].title + '：' ]),
+									E('span', {'class': classname }, [ data[idx][ver].value ]),
+									suppelements
+								])
+							);
+						}
+
+					} else {
+
+						if ('title' === ver) {
+							container_internet_v6.appendChild(
+								E('p', { 'class': 'mt-2'}, [
+									E('span', {'class': ''}, [ data[idx].title ]),
+								])
+							);
+							continue;
+						}
+
+						if (visible) {
+							container_internet_v6.appendChild(
+								E('p', {'class': 'mt-2'}, [
+									E('span', {'class': ''}, [data[idx][ver].title + '：']),
+									E('span', {'class': classname}, [data[idx][ver].value]),
+									suppelements
+								])
+							);
+						}
+					}
+				}
+			}
+
+			container_item.appendChild(container_internet_v4);
+			container_item.appendChild(container_internet_v6);
+		} else {
+			for(var idx in data) {
+				container_item.appendChild(
+					E('p', { 'class': 'mt-2'}, [
+						E('span', {'class': ''}, [ data[idx].title + '：' ]),
+						E('span', {'class': ''}, [ data[idx].value ])
+					])
+				);
+			}
+		}
+
+		container_box.appendChild(container_item);
+		container_box.appendChild(E('hr'));
+		container_wapper.appendChild(container_box);
+		return container_wapper;
+	},
+
+	renderUpdateWanData: function(data, v6) {
+
+		for (var i = 0; i < data.length; i++) {
+			var ifc = data[i];
+
+			if (v6) {
+				this.params.internet.v6.ipprefixv6.value =  ifc.getIP6Prefix() || '-';
+				this.params.internet.v6.gatewayv6.value =  ifc.getGateway6Addr() || '-';
+				this.params.internet.v6.protocol.value=  ifc.getI18n() || E('em', _('Not connected'));
+				this.params.internet.v6.addrsv6.value = ifc.getIP6Addrs() || [ '-' ];
+				this.params.internet.v6.dnsv6.value = ifc.getDNS6Addrs() || [ '-' ];
+
+				if (ifc.getIPAddr()) {
```

Comment:
To check that the interface is "connected", you should use https://openwrt.github.io/luci/jsapi/LuCI.network.Protocol.html#isUp

Reply:
Thank you for reminding.

Can I use ifc.isUp() || ifc.getIP6Addr() to verify v6 network connectivity?

ifc.isUp() || ifc.getIPAddr()　verify the v4 network.

Thank

Reply:
Why can't you just use isUp()?

Reply:
Because I am not sure whether "running" means there is an existing network connection?

Reply:
Actually, I need to know whether the interface is up and whether it is connected to the public network.

Reply:
The state (up/down) of a protocol is more than just about the state of the device. Each protocol such as DHCP or PPP sets the state to "up" only when it has reached a good state (for instance obtained an IP address, but it could be something else).  See [documentation](https://git.openwrt.org/?p=project/netifd.git;a=blob;f=DESIGN;h=4d58bcff5f8b67eef0b3cd253ccec4b08180e071;hb=HEAD#l94).

That means you must only use `isUp()` and trust it to do the right thing, instead of trying to check something else like IP addresses manually.

Reply:
Thank you for your reply

Reply:
Can you fix the code?

Reply:
I have changed the last commit

Reply:
I still see the old code which checks for IP address, did you push your changes?

Reply:
Of course! I just pushed the last version of the code.